### PR TITLE
TDI-40364 tChangeFileEncoding fr

### PR DESCRIPTION
**Pull request by bgunics-talend**
**What is the current behavior?** (You can also link to an open issue here)
TDI-40364

**What is the new behavior?**
TDI-40364

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.designer.components.localprovider/components/tChangeFileEncoding/tChangeFileEncoding_main.javajet [hunk 1]
```diff
@@ -40,15 +41,16 @@ java.io.File outFile_<%=cid%> = new java.io.File(sOutFileN_<%=cid%>);
...
   new java.io.OutputStreamWriter(
     new java.io.FileOutputStream(sOutFileN_<%=cid%>, false), <%=sOutEncoding%>
-));
+)
+  ,<%=bufferSize%>);
     
 java.io.BufferedReader in_<%=cid %> = new java.io.BufferedReader(
   new java.io.InputStreamReader(
     new java.io.FileInputStream(sOriFileN_<%=cid%>), <%=bNeedEncode ? sInEncoding : "System.getProperty(\"file.encoding\")"%>
 ));
```
Comment (clesaec) [inline main/plugins/org.talend.designer.components.localprovider/components/tChangeFileEncoding/tChangeFileEncoding_main.javajet:50]: Why not use here [BufferedReader(inputStream, int) constructor](https://docs.oracle.com/javase/8/docs/api/java/io/BufferedReader.html#BufferedReader-java.io.Reader-int-) to use bufferSize.

File: main/plugins/org.talend.designer.components.localprovider/components/tChangeFileEncoding/tChangeFileEncoding_main.javajet [hunk 2]
```diff
@@ -40,15 +41,16 @@ java.io.File outFile_<%=cid%> = new java.io.File(sOutFileN_<%=cid%>);
...
 
 //8192: the default buffer size of BufferedReader
-char[] cbuf_<%=cid %> = new char[8192];  
+char[] cbuf_<%=cid %> = new char[<%=bufferSize%>];
```
Comment (clesaec) [inline main/plugins/org.talend.designer.components.localprovider/components/tChangeFileEncoding/tChangeFileEncoding_main.javajet:53]: Not really needed here, with BufferedReader

**Review by clesaec — COMMENTED** [review]
Just a comment on BufferedReader and local char array.

Except that, it's ok for me.

**Comment by bgunics-talend** [comment]
Since this pull request has been reviewed and added by @clesaec  I close this pull request.

**Pull request closed without merging**